### PR TITLE
factor out consistency check

### PR DESCRIPTION
**Pull request by polgreen**

### CODE REVIEW

File: src/theory/quantifiers/oracle_engine.h [hunk 1]
```diff
@@ -61,6 +61,12 @@ class OracleEngine : public QuantifiersModule
   /** Identify. */
   std::string identify() const override;
 
+  /* check pairs of nodes are consistent */
```
Comment (ajreynol) [inline src/theory/quantifiers/oracle_engine.h:64]: Can you move the documentation to here?

File: src/theory/quantifiers/oracle_engine.cpp [hunk 2]
```diff
@@ -71,6 +77,30 @@ void OracleEngine::reset_round(Theory::Effort e) {}
 
 void OracleEngine::registerQuantifier(Node q) {}
 
+
+// need to know the oracle
+bool OracleEngine::checkConsistent(
+  const std::vector< std::pair<Node, Node> >& ioPairs, // input-output pairs from the oracle
```
Comment (ajreynol) [inline src/theory/quantifiers/oracle_engine.cpp:83]: Nit: move these sentences to the documentation in the h.

File: src/theory/quantifiers/oracle_engine.cpp [hunk 3]
```diff
@@ -71,6 +77,30 @@ void OracleEngine::reset_round(Theory::Effort e) {}
...
+  std::vector<Node> &apps, std::vector<Node>& lemmas) // lemmas is to be filled, if the input-output pairs are inconsistent with model
+  {
+    bool consistent=true;
+    NodeManager* nm = NodeManager::currentNM();
+    eq::EqualityEngine* eq = getEqualityEngine();
+    for(unsigned int i=0; i<ioPairs.size(); i++)
```
Comment (ajreynol) [inline src/theory/quantifiers/oracle_engine.cpp:89]:
```suggestion
    for(size_t i=0; i<ioPairs.size(); i++)
```

File: src/theory/quantifiers/oracle_engine.cpp [hunk 4]
```diff
@@ -71,6 +77,30 @@ void OracleEngine::reset_round(Theory::Effort e) {}
...
+    {
+      // get predicted result
+      Node predictedResult = eq->getRepresentative(apps[i]);
```
Comment (ajreynol) [inline src/theory/quantifiers/oracle_engine.cpp:92]: Actually, this isn't what I had wanted.

I wanted the input to contain the expected values; the method should call the oracles.

**Review by ajreynol — CHANGES_REQUESTED** [review]

File: src/theory/quantifiers/oracle_engine.cpp (hunk 4 above)
Comment (polgreen) [reply src/theory/quantifiers/oracle_engine.cpp:92]: So std::vector<Node, Node> ioPairs is the inputs (symbolic) and expected outputs? Or is it the inputs (values) and expected outputs?

Comment (ajreynol) [reply src/theory/quantifiers/oracle_engine.cpp:92]: An example pair would be `f(5,6), 7`, return true iff the oracle thinks f(5,6)=7

Comment (polgreen) [reply src/theory/quantifiers/oracle_engine.cpp]: done

**Review by ajreynol — APPROVED** [review]

**Pull request merged**